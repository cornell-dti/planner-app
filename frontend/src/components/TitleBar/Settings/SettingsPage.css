.SettingsSection {
  margin-top: 10px;
}

p.SettingsSectionTitle {
  vertical-align: top;
  font-size: 16px;
  padding-top: 20px;
}

.SettingsSectionContent {
  vertical-align: top;
}

@media only screen and (min-width: 768px) {
  p.SettingsSectionTitle {
    width: 200px;
    display: inline-block;
  }

  .SettingsSectionContent {
    display: inline-block;
    width: calc(100% - 200px);
  }
}

.SettingsButton {
  padding-top: 15px;
}

.SettingsButton button {
  color: white;
  border: 1px white solid;
  border-radius: 5px;
  padding: 5px 10px;
  line-height: 14px;

  margin-top: 5px;
}

.SignButton {
  background: #e93838 !important;
  border: none !important;
  border-radius: 20px !important;
  font-size: 16px !important;
  padding: 10px 20px !important;
}

.ColorConfigItemList {
  display: inline-block;
  width: 100%;
  margin: 0;
  padding: 0;
  list-style: none;
}
