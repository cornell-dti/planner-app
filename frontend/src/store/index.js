// @flow strict

import { createStore } from 'redux';
import type { Store } from 'redux';
import rootReducer from './reducers';
import type { State } from './store-types';
import type { Action } from './action-types';

const store: Store<State, $Subtype<Action>> = createStore(rootReducer);
export default store;
