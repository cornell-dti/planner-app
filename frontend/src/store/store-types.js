// @flow

export type SubTask = {|
  +name: string; // Example: "SubTask 1 Name"
  +id: number; // Example: 32432
  +complete: boolean;
|};

/**
 * The task in the store.
 */
export type Task = {|
  +name: string; // Example: "Task 1 name"
  +id: number; // Example: 3213
  +tag: string; // Example: "CS 2112"
  +date: Date; // Example: new Date()
  +complete: boolean;
  +subtaskArray: SubTask[];
|};

/**
 * The tag color picker maps a tag to a color.
 */
export type TagColorConfig = {
  +[tag: string]: string
};

/**
 * The type of the entire redux state.
 */
export type State = {
  +mainTaskArray: Task[],
  +tagColorPicker: TagColorConfig,
  +bearStatus: string
};
