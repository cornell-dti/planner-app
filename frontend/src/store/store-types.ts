import { Map, Set } from 'immutable';

export type Tag = {
  readonly id: string;
  readonly order: number;
  readonly name: string;
  readonly color: string;
  readonly classId: string | null;
};

export type SubTask = {
  readonly id: string;
  readonly order: number;
  readonly name: string; // Example: "SubTask 1 Name"
  readonly complete: boolean;
  readonly inFocus: boolean; // Whether the subtask is in focus
};

export type SubTaskWithoutId = Pick<SubTask, 'order' | 'name' | 'complete' | 'inFocus'>;
export type SubTaskWithoutIdOrder = Pick<SubTask, 'name' | 'complete' | 'inFocus'>;
/**
 * The subtask type without id order and with every field as optional.
 */
export type PartialSubTask = Partial<SubTaskWithoutIdOrder>;

export type CommonTask = {
  readonly id: string;
  readonly order: number;
  readonly name: string; // Example: "Task 1 name"
  readonly tag: string; // ID of the tag
  readonly date: Date; // Example: new Date()
  readonly complete: boolean;
  readonly inFocus: boolean; // Whether the task is in focus
  readonly children: Set<string>;
};

export type LegacyTask = CommonTask;

export type OneTimeTask = CommonTask & {
  readonly type: 'ONE_TIME';
};

/**
 * The task type without id and every field optional.
 * Fields that are filled represent differences from master template
 */

export type PartialTask = Partial<Pick<CommonTask, MainTaskProperties | 'children' >>;

export type RepeatingPattern =
  | { readonly type: 'WEEKLY'; readonly bitSet: number /* 7-bit */ }
  | { readonly type: 'BIWEEKLY'; readonly bitSet: number /* 14-bit */ }
  | { readonly type: 'MONTHLY'; readonly bitSet: number /* 31-bit */ };

export type RepeatMetaData = {
  readonly startDate: Date;
  readonly endDate: Date | null;
  readonly pattern: RepeatingPattern;
};

export type ForkedTaskMetaData = PartialTask & {
  readonly forkId: string | null;
  readonly replaceDate: Date;
};

export type RepeatingTask = CommonTask & {
  readonly type: 'MASTER_TEMPLATE';
  readonly repeats: RepeatMetaData;
  readonly forks: readonly ForkedTaskMetaData[];
};

export type Task = LegacyTask | OneTimeTask | RepeatingTask;

type MainTaskProperties = 'order' | 'name' | 'tag' | 'date' | 'complete' | 'inFocus';
/**
 * The task type without id and subtask.
 */
export type MainTask = Readonly<Pick<CommonTask, MainTaskProperties>>;
/**
 * The task type without id and subtask, and with all properties as optional.
 */
export type PartialMainTask = Partial<MainTask>;

/**
 * The type of user settings.
 */
export type Settings = {
  readonly completedOnboarding: boolean;
  readonly theme: 'light' | 'dark';
};

export type BannerMessageIds =
  | '2019-03-10-quota-exceeded-incident';

export type BannerMessageStatus = {
  readonly [I in BannerMessageIds]?: boolean;
};

/**
 * The type of a course info entry.
 */
export type Course = {
  readonly courseId: number;
  readonly subject: string;
  readonly courseNumber: string;
  readonly title: string;
  readonly examTimes: { readonly type: 'final' | 'prelim'; readonly time: number }[];
};

/**
 * The type of the entire redux state.
 */
export type State = {
  readonly tags: Map<string, Tag>;
  readonly tasks: Map<string, Task>;
  readonly dateTaskMap: Map<string, Set<string>>;
  readonly subTasks: Map<string, SubTask>;
  readonly taskChildrenMap: Map<string, Set<string>>;
  readonly settings: Settings;
  readonly bannerMessageStatus: BannerMessageStatus;
  readonly courses: Map<string, Course[]>;
};
