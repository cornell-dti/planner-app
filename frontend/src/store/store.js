// @flow strict

import { createStore } from 'redux';
import type { Store } from 'redux';
import rootReducer from './reducers';
import type { State } from './store-types';
import type { Action } from './action-types';
import type { AppUser } from '../util/firebase-util';
import { error } from '../util/general-util';

export type GlobalStore = Store<State, $Subtype<Action>>;

const store: GlobalStore = createStore(rootReducer);
let appUser: AppUser | null = null;

/**
 * Initialize the app and record a global user and returns the initialized store.
 *
 * @param {AppUser} user the user of the app.
 * @return {GlobalStore} the newly initialized store.
 */
export function initializeApp(user: AppUser): GlobalStore {
  appUser = user;
  return store;
}

/**
 * Returns the global app user.
 *
 * @return {AppUser} the global app user.
 */
export const getAppUser = (): AppUser => appUser ?? error('App is not initialized.');

/**
 * Dispatch an action and returns the dispatched action.
 *
 * @param {Action} action the action to dispatch.
 * @return {Action} the action to dispatch.
 */
export const dispatchAction = (action: Action): Action => store.dispatch(action);
