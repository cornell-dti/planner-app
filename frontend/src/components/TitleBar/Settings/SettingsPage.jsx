// @flow strict

import * as React from 'react';
import type { Node } from 'react';
import TagItem from '../Tags/TagItem';
import ClassTagAdder from '../Tags/ClassTagAdder';
import OtherTagAdder from '../Tags/OtherTagAdder';
import styles from './SettingsPage.css';
import type { State, Tag } from '../../../store/store-types';
import { firebaseSignOut } from '../../../util/firebase-util';
import { simpleConnect } from '../../../store/react-redux-util';

type Props = {| +classTags: Tag[]; +otherTags: Tag[] |};

/**
 * The class adder component.
 *
 * @return {Node} rendered component.
 * @constructor
 */
const ClassAdder = (): Node => (
  <div className={styles.SettingsSection}>
    <p className={styles.SettingsSectionTitle}>Add Classes</p>
    <div className={styles.SettingsSectionContent}>
      <ClassTagAdder />
      <small>Add classes to get upcoming prelims automatically added to your planner.</small>
    </div>
  </div>
);

/**
 * The exam exporter component.
 *
 * @return {Node} rendered component.
 * @constructor
 */
const ExamImporter = (): Node => (
  <div className={styles.SettingsSection}>
    <p className={styles.SettingsSectionTitle}>Auto Import Exams</p>
    <div className={`${styles.SettingsButton} ${styles.SettingsSectionContent}`}>
      Click the following button to automatically import the prelims and finals
      from your registered classes into your planner.
      <br />
      <button type="button">Import</button>
    </div>
  </div>
);

/**
 * The sign out component.
 *
 * @return {Node} rendered component.
 * @constructor
 */
const SignOut = (): Node => (
  <div className={styles.SettingsSection}>
    <div className={styles.SettingsButton}>
      <button type="button" className={styles.SignButton} onClick={firebaseSignOut}>
        Sign Out
      </button>
    </div>
  </div>
);

/**
 * The tags container component.
 *
 * @param {string} title the title of the tags.
 * @param {Node} children the tags as children.
 * @return {Node} rendered component.
 * @constructor
 */
const TagsContainer = ({ title, children }: {| +title: string; +children: Node |}): Node => (
  <div className={styles.SettingsSection}>
    <p className={styles.SettingsSectionTitle}>{title}</p>
    <div className={styles.SettingsSectionContent}>
      <ul className={styles.ColorConfigItemList}>
        {children}
      </ul>
    </div>
  </div>
);

/**
 * Render an array of tags.
 *
 * @param {Tag[]} tags an array of tags.
 * @return {Node} rendered tags.
 */
const renderTags = (tags: Tag[]): Node => tags.map(t => <TagItem key={t.id} tag={t} />);

/**
 * The settings page.
 *
 * @param {Tag[]} classTags all the class tags.
 * @param {Tag[]} otherTags all the other tags.
 * @return {Node} the rendered editor.
 * @constructor
 */
function SettingsPage({ classTags, otherTags }: Props): Node {
  return (
    <div>
      <ClassAdder />
      <TagsContainer title="Class Tags">
        {renderTags(classTags)}
      </TagsContainer>
      <ExamImporter />
      <TagsContainer title="Other Tags">
        {renderTags(otherTags)}
        <OtherTagAdder />
      </TagsContainer>
      <SignOut />
    </div>
  );
}

const ConnectedSettingsPage = simpleConnect<{||}, Props>(
  ({ tags }: State) => {
    const classTags = [];
    const otherTags = [];
    tags.forEach((tag) => {
      if (tag.type === 'class') {
        classTags.push(tag);
      } else if (tag.name !== 'None') {
        otherTags.push(tag);
      }
    });
    return { classTags, otherTags };
  },
)(SettingsPage);
export default ConnectedSettingsPage;
