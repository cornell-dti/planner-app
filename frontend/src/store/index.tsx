import React, { ReactElement } from 'react';
import { Provider as ReactReduxProvider } from 'react-redux';
import { store, storeForTesting } from './store';

type Props = { readonly children: ReactElement };

export const Provider = ({ children }: Props): ReactElement => (
  <ReactReduxProvider store={store}>{children}</ReactReduxProvider>
);

export const ProviderForTesting = ({ children }: Props): ReactElement => (
  <ReactReduxProvider store={storeForTesting}>{children}</ReactReduxProvider>
);
