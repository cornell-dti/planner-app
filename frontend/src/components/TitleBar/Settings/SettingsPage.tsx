import React, { ReactElement, ReactNode } from 'react';
import { Map } from 'immutable';
import { connect } from 'react-redux';
import TagItem from '../Tags/TagItem';
import ClassTagAdder from '../Tags/ClassTagAdder';
import OtherTagAdder from '../Tags/OtherTagAdder';
import styles from './SettingsPage.module.css';
import { Tag, State } from '../../../store/store-types';
import { completeOnboarding, importCourseExams } from '../../../firebase/actions';
import { firebaseSignOut } from '../../../firebase/auth';

/**
 * The class adder component.
 *
 * @return {Node} rendered component.
 * @constructor
 */
const ClassAdder = (): ReactElement => (
  <div className={styles.SettingsSection}>
    <p className={styles.SettingsSectionTitle}>Add Classes</p>
    <div className={styles.SettingsSectionContent}>
      <ClassTagAdder />
      <small>Add classes to get upcoming prelims automatically added to your planner.</small>
    </div>
  </div>
);

const ExamImporter = (): ReactElement => (
  <div className={styles.SettingsSection}>
    <p className={styles.SettingsSectionTitle}>Auto Import Exams</p>
    <div className={`${styles.SettingsButton} ${styles.SettingsSectionContent}`}>
      Click the following button to automatically import the prelims and finals
      from your added classes into your planner. We will only import those that
      appears on Cornell prelim/final schedule webpage.
      <br />
      <button type="button" onClick={importCourseExams}>Import</button>
    </div>
  </div>
);

type TagsContainerProps = { readonly title: string; readonly children: ReactNode };

/**
 * The tags container component.
 *
 * @param {string} title the title of the tags.
 * @param {Node} children the tags as children.
 * @return {Node} rendered component.
 * @constructor
 */
const TagsContainer = ({ title, children }: TagsContainerProps): ReactElement => (
  <div className={styles.SettingsSection}>
    <p className={styles.SettingsSectionTitle}>{title}</p>
    <div className={styles.SettingsSectionContent}>
      <ul className={styles.ColorConfigItemList}>
        {children}
      </ul>
    </div>
  </div>
);

type Props = { readonly tags: Map<string, Tag> };

/**
 * The settings page.
 */
function SettingsPage({ tags }: Props): ReactElement {
  const classTags: Tag[] = [];
  const otherTags: Tag[] = [];
  tags.forEach((tag) => {
    if (tag.classId !== null) {
      classTags.push(tag);
    } else if (tag.name !== 'None') {
      otherTags.push(tag);
    }
  });
  const renderTags = (arr: Tag[]): ReactNode => arr.map((tag: Tag) => (
    <TagItem key={tag.id} tag={tag} />
  ));
  return (
    <div>
      <ClassAdder />
      <TagsContainer title="Class Tags">
        {renderTags(classTags)}
      </TagsContainer>
      <ExamImporter />
      <TagsContainer title="Other Tags">
        {renderTags(otherTags)}
        <OtherTagAdder />
      </TagsContainer>
      <div className={styles.FinalRowButtonContainer}>
        <div className={styles.SettingsButton}>
          <button
            type="button"
            className={[styles.FinalRowButton, styles.SignButton].join(' ')}
            onClick={firebaseSignOut}
          >
            Sign Out
          </button>
        </div>
        <a
          className={styles.FeedbackButton}
          href="https://goo.gl/forms/PZUZ1Ze6kN82EmcD2"
          target="_blank"
          rel="noopener noreferrer"
        >
          Give Feedback
        </a>
        <span className={styles.Padding} />
        <div className={styles.SettingsButton}>
          <button
            type="button"
            className={[styles.FinalRowButton, styles.ReplayTutorialButton].join(' ')}
            onClick={() => completeOnboarding(false)}
          >
            Replay Tutorial
          </button>
        </div>
      </div>
    </div>
  );
}

const Connected = connect(({ tags }: State) => ({ tags }))(SettingsPage);
export default Connected;
