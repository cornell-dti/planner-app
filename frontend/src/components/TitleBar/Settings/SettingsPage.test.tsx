import React from 'react';
import { Map } from 'immutable';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { Tag } from 'store/store-types';
import { store } from 'store/store';
import { ClassAdder, ExamImporter, TagsContainer, SettingsPage } from './SettingsPage';

it('ClassAdder matches snapshot.', () => {
  const tree = renderer.create(<Provider store={store}><ClassAdder /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('ExamImporter matches snapshot.', () => {
  const tree = renderer.create(<Provider store={store}><ExamImporter /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('TagsContainer matches snapshot.', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <TagsContainer title="TITLE">TEST</TagsContainer>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('SettingsPage matches snapshot.', () => {
  const tags = Map<string, Tag>()
    .set('t1', { id: 't1', order: 1, name: 'T1', color: 'red', classId: null })
    .set('t2', { id: 't2', order: 2, name: 'CS2112: Critter', color: 'red', classId: 'classId' });
  const tree = renderer.create(
    <Provider store={store}>
      <SettingsPage tags={tags} />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
