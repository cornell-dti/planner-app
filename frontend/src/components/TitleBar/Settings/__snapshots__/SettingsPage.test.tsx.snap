// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassAdder matches snapshot. 1`] = `
<div
  className="SettingsSection"
>
  <p
    className="SettingsSectionTitle"
  >
    Add Classes
  </p>
  <div
    className="SettingsSectionContent"
  >
    <div
      className="TagColorConfigItemAdder SearchClasses"
      title="Search for a class"
    >
      <div>
        <input
          className="SearchInput"
          onChange={[Function]}
          placeholder="Search for classes (e.g. CS 2110, Introduction to Creative Writing)"
          type="text"
          value=""
        />
        <div />
      </div>
    </div>
    <small>
      Add classes to get upcoming prelims automatically added to your planner.
    </small>
  </div>
</div>
`;

exports[`ExamImporter matches snapshot. 1`] = `
<div
  className="SettingsSection"
>
  <p
    className="SettingsSectionTitle"
  >
    Auto Import Exams
  </p>
  <div
    className="SettingsButton SettingsSectionContent"
  >
    Click the following button to reimport the prelims and finals from your classes into your planner. We will only import those that appears on Cornell prelim/final schedule webpage.
    <br />
    <button
      onClick={[Function]}
      tabIndex={0}
      title="Reimport Exams"
      type="button"
    >
      Reimport Exams
    </button>
  </div>
</div>
`;

exports[`SettingsPage matches snapshot. 1`] = `
<div>
  <div
    className="SettingsSection"
  >
    <p
      className="SettingsSectionTitle"
    >
      Add Classes
    </p>
    <div
      className="SettingsSectionContent"
    >
      <div
        className="TagColorConfigItemAdder SearchClasses"
        title="Search for a class"
      >
        <div>
          <input
            className="SearchInput"
            onChange={[Function]}
            placeholder="Search for classes (e.g. CS 2110, Introduction to Creative Writing)"
            type="text"
            value=""
          />
          <div />
        </div>
      </div>
      <small>
        Add classes to get upcoming prelims automatically added to your planner.
      </small>
    </div>
  </div>
  <div
    className="SettingsSection"
  >
    <p
      className="SettingsSectionTitle"
    >
      Class Tags
    </p>
    <div
      className="SettingsSectionContent"
    >
      <ul
        className="ColorConfigItemList"
      >
        <li
          className="ColorConfigItem"
        >
          <span
            style={
              Object {
                "display": "inline-block",
                "width": "calc(100% - 175px)",
              }
            }
          >
            <span
              className="TagName"
            >
              CS2112
            </span>
            <span
              className="ClassExpandedTitle"
            >
              Critter
            </span>
          </span>
          <button
            className="ColorEdit"
            onClick={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="ColorDisplay"
              style={
                Object {
                  "backgroundColor": "red",
                }
              }
            />
            <span
              title="More"
            >
              <svg
                alt="More"
                className="Arrow"
                height="1em"
                onKeyUp={[Function]}
                tabIndex={0}
                title="More"
                width="1em"
              >
                v.svg
              </svg>
            </span>
          </button>
          <button
            className="DeleteTag"
            onClick={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              title="Delete"
            >
              <svg
                alt="Delete"
                height="1em"
                onKeyUp={[Function]}
                tabIndex={0}
                title="Delete"
                width="1em"
              >
                XLight.svg
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="SettingsSection"
  >
    <p
      className="SettingsSectionTitle"
    >
      Auto Import Exams
    </p>
    <div
      className="SettingsButton SettingsSectionContent"
    >
      Click the following button to reimport the prelims and finals from your classes into your planner. We will only import those that appears on Cornell prelim/final schedule webpage.
      <br />
      <button
        onClick={[Function]}
        tabIndex={0}
        title="Reimport Exams"
        type="button"
      >
        Reimport Exams
      </button>
    </div>
  </div>
  <div
    className="SettingsSection"
  >
    <p
      className="SettingsSectionTitle"
    >
      Other Tags
    </p>
    <div
      className="SettingsSectionContent"
    >
      <ul
        className="ColorConfigItemList"
      >
        <li
          className="ColorConfigItem"
        >
          <span
            className="TagName"
            style={
              Object {
                "width": "calc(100% - 175px)",
              }
            }
          >
            <input
              className="TagEdit"
              onChange={[Function]}
              type="text"
              value="T1"
            />
          </span>
          <button
            className="ColorEdit"
            onClick={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="ColorDisplay"
              style={
                Object {
                  "backgroundColor": "red",
                }
              }
            />
            <span
              title="More"
            >
              <svg
                alt="More"
                className="Arrow"
                height="1em"
                onKeyUp={[Function]}
                tabIndex={0}
                title="More"
                width="1em"
              >
                v.svg
              </svg>
            </span>
          </button>
          <button
            className="DeleteTag"
            onClick={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              title="Delete"
            >
              <svg
                alt="Delete"
                height="1em"
                onKeyUp={[Function]}
                tabIndex={0}
                title="Delete"
                width="1em"
              >
                XLight.svg
              </svg>
            </span>
          </button>
        </li>
        <li
          className="ColorConfigItem"
        >
          <input
            className="TagName AddTagName"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="New Tag"
            type="text"
            value=""
          />
          <button
            className="ColorEdit"
            onClick={[Function]}
            tabIndex={0}
            type="button"
          >
            Blue
            <span
              className="ColorDisplay"
              style={
                Object {
                  "backgroundColor": "#289de9",
                }
              }
            />
            <span
              title="More"
            >
              <svg
                alt="More"
                className="Arrow"
                height="1em"
                onKeyUp={[Function]}
                tabIndex={0}
                title="More"
                width="1em"
              >
                v.svg
              </svg>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="SettingsSection"
  >
    <p
      className="SettingsSectionTitle"
    >
      Canvas Calendar
    </p>
    <div
      className="SettingsSectionContent"
    >
      <div
        style={
          Object {
            "display": "block",
          }
        }
      >
        <form
          className="CalendarForm SettingsButton"
          onSubmit={[Function]}
        >
          <input
            className="CalendarInput"
            onChange={[Function]}
            placeholder="Paste your Canvas iCal link here"
            type="text"
            value=""
          />
          <button
            title="Link Canvas iCal"
            type="submit"
          >
             Save 
          </button>
        </form>
        <a
          className="HelpButton"
          href="https://community.canvaslms.com/docs/DOC-10691-4212717348"
          title="Link to Canvas iCal guide"
        >
          Having trouble finding the iCal link?
        </a>
      </div>
      <div
        className="SettingsButton"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <p
          style={
            Object {
              "wordBreak": "break-all",
            }
          }
        >
          Your Canvas calendar feed is linked.
          <br />
        </p>
        <button
          onClick={[Function]}
          title="Remove Canvas iCal Link"
          type="button"
        >
          Unlink
        </button>
      </div>
    </div>
  </div>
  <div
    className="FinalRowButtonContainer"
  >
    <div
      className="SettingsButton"
    >
      <button
        className="FinalRowButton SignButton"
        onClick={[Function]}
        tabIndex={0}
        title="Sign out of Samwise"
        type="button"
      >
        Sign Out
      </button>
    </div>
    <a
      className="FeedbackButton"
      href="https://goo.gl/forms/PZUZ1Ze6kN82EmcD2"
      rel="noopener noreferrer"
      tabIndex={0}
      target="_blank"
      title="Link to Feedback form"
    >
      Give Feedback
    </a>
    <span
      className="Padding"
    />
    <div
      className="SettingsButton"
    >
      <button
        className="FinalRowButton ReplayTutorialButton"
        onClick={[Function]}
        tabIndex={0}
        title="Click to replay tutorial"
        type="button"
      >
        Replay Tutorial
      </button>
    </div>
  </div>
</div>
`;

exports[`TagsContainer matches snapshot. 1`] = `
<div
  className="SettingsSection"
>
  <p
    className="SettingsSectionTitle"
  >
    TITLE
  </p>
  <div
    className="SettingsSectionContent"
  >
    <ul
      className="ColorConfigItemList"
    >
      TEST
    </ul>
  </div>
</div>
`;
